src/ripple/basics/impl/ResolverAsio.cpp

```diff
@@ -33,9 +32,75 @@
 
 namespace ripple {
 
+/** Mix-in to track when all pending I/O is complete.
+    Derived classes must be callable with this signature:
+        void asyncHandlersComplete()
+*/
+template <class Derived>
+class AsyncObject
+{
+protected:
+    AsyncObject ()
+        : m_pending (0)
+    { }
+
+public:
+    ~AsyncObject ()
+    {
+        // Destroying the object with I/O pending? Not a clean exit!
+        assert (m_pending.load () == 0);
```

Comment:
This class needs to go away - it's only used for one thing (the resolver). I wouldn't invest any time on fixing it up.